modules/post/osx/capture/keylog_recorder.rb

```diff
@@ -62,9 +62,9 @@ def run_ruby_code
       :duration => datastore['DURATION'].to_i,
       :port => self.port
     }
-    cmd = ['ruby', '-e', ruby_code(opts)]
 
-    rpid = cmd_exec(cmd.shelljoin, nil, 10)
+    cmd = "OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES ruby -e #{ruby_code(opts).shellescape}"
```

Comment:
The TL;DR here is that High Sierra doesn't allow forking anymore, but we can work around it by setting an environment variable. More info here: https://blog.phusion.nl/2017/10/13/why-ruby-app-servers-break-on-macos-high-sierra-and-what-can-be-done-about-it/